api/metrics/rent_burdened.py

```diff
@@ -0,0 +1,82 @@
+from api.utils.database import rows_to_dicts
+
+
+class RentBurdenedMetrics:
+    """
+    Metrics for rent burdened households data.
+    """
+
+    def __init__(self, con):
+        self.con = con
+
+    def rent_burdened(self, year, segment):
+        """
+        Returns the percent of households who are rent burdened in the area,
+        and the area number.
+        Args:
+            year (int): period ending year to filter by
+            segment (str): population segment to filter by
+        """
+        query = """
+        SELECT
+            area_number,
+            CAST(value AS INTEGER) AS value
+        FROM rent_burdened_households
+        WHERE period_end_year == {year}
+        AND segment == "{segment}"
+        """.format(year=year, segment=segment)
+        cur = self.con.cursor()
+        cur.execute(query)
+        rows = rows_to_dicts(cur, cur.fetchall())
+        return rows
+        
+    def get_max_burdened(self):
+        """
+        Returns the highest percentage of rent burdened households in an area,
+        its area number, and the period.
+        """
+        query = """
+        SELECT
+            max(ROUND(value,2)) AS value,
+            area_number,
+            period_end_year
+        FROM rent_burdened_households
+        """
+        cur = self.con.cursor()
+        cur.execute(query)
+        rows = rows_to_dicts(cur, cur.fetchall())
+        return rows
+    
+    def get_min_burdened(self):
+        """
+        Returns the smallest percentage of rent burdened households in an area,
+        its area number, and the period end year.
+        """
+        query = """
+        SELECT
+            min(ROUND(value,2)) AS value,
+            area_number,
+            period_end_year
+        FROM rent_burdened_households
+        """
+        cur = self.con.cursor()
+        cur.execute(query)
+        rows = rows_to_dicts(cur, cur.fetchall())
+        return rows
+    
+    def average_burden_area(self):
```

Comment:
This is an interesting idea for a metric. Averaging percentages might be misleading because each percentage represents a different year with a different denominator, so we don't have a way to weight the averages. For example, suppose that between two years, the number of rent burdened households stays the same, but in the second year, some new houses are finished and non rent burdened people move in. The denominator of total households increases, causing the percentage of rent burdened to go down, and causing the average to go down.

You don't have to change it, but just wanted to point this out.